bin/lib/database.js

```diff
@@ -77,24 +77,82 @@ function readFromDatabase(item, table){
 }
 
 function scanDatabase(query){
+	
+	debug('Scanning database', query);
 
-	return new Promise( (resolve, reject) => {
+	const results = [];
 
-		if(query.TableName === undefined || query.TableName === null){
-			reject(`'TableName' argument is ${query.TableName}`);
-		} else {
-			
-			DynamoClient.scan(query, function(err, data){
+	function scan(query){
 
-				if(err){
-					reject(err);
-				} else {
-					resolve(data);
-				}
+		return new Promise( (resolve, reject) => {
 
-			})
+			if(query.TableName === undefined || query.TableName === null){
+				reject(`'TableName' argument is ${query.TableName}`);
+			} else {
 
-		}
+				DynamoClient.scan(query, function(err, data){
+
+					if(err){
+						reject(err);
+					} else {
+						debug(data.Items.length);
+						results.push(data);
+						if(data.LastEvaluatedKey !== undefined){
```

Comment:
I disagree (but only based on an opinion). The way I read the code is:

If [bad thing]:
  handleBadThing()
else:
  //behave as expected.